schedpolicy/sched_policy.go

```diff
@@ -0,0 +1,52 @@
+//go:generate mockgen -package=mock -destination=mock/schedpolicy.mock.go github.com/libopenstorage/openstorage/schedpolicy SchedulePolicy
+package schedpolicy
+
+import "errors"
+
+var (
+	ErrNotImplemented = errors.New("Not Implemented")
+)
+
+type SchedulePolicy interface {
+	// SchedPolicyCreate creates a policy with given name and schedule.
+	SchedPolicyCreate(name, sched string) error
+	// SchedPolicyUpdate updates a policy with given name and schedule.
+	SchedPolicyUpdate(name, sched string) error
+	// SchedPolicyDelete deletes a policy with given name.
+	SchedPolicyDelete(name string) error
+	// SchedPolicyEnumerate enumerates all configured policies or the ones specified.
+	SchedPolicyEnumerate([]string) ([]*SchedPolicy, error)
+}
+
+type Manager struct {
+	SchedulePolicy
+}
+
+func NewSchedulePolicyManager(sched SchedulePolicy) *Manager {
```

Comment:
All of the following is a suggestion. What you have done works, but it is not normal golang so it is not as easy to read:

I'm not sure I understand the point of this function (and for Secret also). NullSchedManager is an implementation of `SchedulePolicy`.  server/cluster.go:36 can be `schedpolicy.SchedulePolicy` the interface instead of a pointer. Then the function below which returns a pointer to a schedule manager struct can be used since it satisfies the interface.

You can also add the following line to the top of the file to make sure that nullSchedMgr always satisfies the interface:

```go
var _ SchedPolicy = &nullSchedMgr{}
```

This will have the compiler immediately check if this struct satisfies the interface.